crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -20,10 +120,43 @@ impl ParsedQProperty {
             let _comma = input.parse::<Token![,]>()?;
             let ident = input.parse()?;
 
-            // TODO: later we'll need to parse setters and getters here
-            // which are key-value, hence this not being parsed as a list
+            if input.is_empty() {
+                // No flags so fill with default options
+                Ok(Self {
+                    ident,
+                    ty,
+                    flags: QPropertyFlags::all_flags(),
+                })
+            } else {
+                let _comma = input.parse::<Token![,]>()?; // Start of final identifiers
+
+                let punctuated_flags: Punctuated<Meta, Token![,]> =
+                    Punctuated::parse_terminated(input)?;
+
+                let flags: Vec<Meta> = punctuated_flags.into_iter().collect(); // Removes the commas while collecting into Vec
 
-            Ok(Self { ident, ty })
+                let mut passed_flags = QPropertyFlags::new();
+
+                let mut flag_strings: Vec<String> = Vec::new();
+
+                for flag in flags {
+                    let flag_string: String = passed_flags.add_from_meta(flag)?;
+                    flag_strings.push(flag_string);
+                }
+
+                if !flag_strings.contains(&String::from("read")) {
+                    return Err(Error::new(
+                        input.span(),
+                        "If flags are passed, read must be explicitly specified",
+                    ));
+                }
```

Comment:
you don't actually need the vector of strings in the end other than for checking is there read in it ? So instead change to a bool and check in the loop.

eg

```rust
let mut found_read = false;

for flag in flags {
    let flag_string: String = passed_flags.add_from_meta(flag)?;
    found_read |= flag_string == "read";
}

if !found_read {
...
```

Reply:
This is still valid as to remove the list and use a bool if QPropertyFlags changes so that read is not an Option so you cannot check if it's been set, see other comment.

Reply:
Code has changed.